src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -37,6 +46,36 @@ std::shared_ptr<ov::Node> change_constant_precision_to_fp16(std::shared_ptr<ov::
             dst_data[i] = static_cast<ov::float16>(src_data[i]);
         }
     }
+
+    // thresholds in percents
+    // if more than 1% of a float is underflowed print debug info about that
+    // if more than 75% of a FP32 constant do not fit into FP16 keep in FP32
+    auto keep_threshold = static_cast<float>(ngraph::getenv_int("KEEP_FP32_THRESHOLD", 75));
+    auto fraction_underflow = 100.0f * static_cast<float>(num_underflow) / static_cast<float>(size);
+
+    //    ATTENTION: need this output for debug purpose, better not to remove
+    //    auto info_threshold = static_cast<float>(ngraph::getenv_int("THRESHOLD_TO_INFORM_FP16_UNDERFLOW", 0.));
+    //    if (fraction_underflow > info_threshold) {
+    //        std::cout << std::setprecision(2) << std::fixed << std::setw(5) << fraction_underflow;
+    //        std::cout << " % underflow on '" << constant->get_friendly_name() << "', ";
+    //        std::cout << "size: " << size;
+    //        if (fraction_underflow > keep_threshold) {
+    //            std::cout << " [KEEP IN FP32]";
+    //        }
+    //        std::cout << std::endl;
+    //    }
+
+    if (fraction_underflow > keep_threshold) {
```

Comment:
This doesn't align with comment earlier. This if will work for every `fraction_underflow` that is above 75% not 1%.

Reply:
Thanks for noticing! In the version comment aligned comment with the code